src/secrets-kubernetes/src/lib.rs

```diff
@@ -106,6 +92,45 @@ impl SecretsController for KubernetesSecretsController {
             )
             .await?;
 
+        // guarantee that all new secrets are reflected on our local filesystem
+        let secrets_storage_path = PathBuf::from("/secrets");
+        for op in ops.iter() {
+            match op {
+                SecretOp::Ensure { id, .. } => {
+                    let file_path = secrets_storage_path.join(format!("{}", id));
+                    if !poll_until_success(|| Path::exists(&*file_path), POLL_TIMEOUT).await {
+                        error!("Secret write operation has timed out. Secret with id {} could not be written", id);
+                        return Err(anyhow!("Secret write operation has timed out. Secret with id {} could not be written", id));
+                    }
+                }
+                SecretOp::Delete { id } => {
+                    let file_path = secrets_storage_path.join(format!("{}", id));
+                    if !poll_until_success(|| !Path::exists(&*file_path), POLL_TIMEOUT).await {
+                        error!("Secret delete operation has timed out. Secret with id {} could not be written", id);
+                        return Err(anyhow!("Secret delete operation has timed out. Secret with id {} could not be written", id));
+                    }
```

Comment:
We can skip the poll on delete, since a stale secret file won't be visible to users.